ssrando/ssrando
# Starting Item Options

Aims:
* Add additional start items that were missed first time round (extra wallets and crystal packs)
* Add options to start with max bugs and treasures
* Allow starting with a full wallet
* Add option for wallet upgrades to come pre-filled with Rupees
* Add initial options for starting with a shield and bottles (not how I envision this looking in the future but, as a highly requested feature, it's good solution for now)

Details:
* The additional items work the same as the others. Crystal packs are chosen with a spin box as discussed
* Starting: rupees, health, interface choice, bugs, treasures, shield, and bottles are stored in a bitfield at the end of the startflags (this was already *kinda* the case for the interface choice)
* Starting shield and bottles are options separate from the main item list (this is due to their "temporary" status)
  * These options are "temporary" as we hope to create a popout dialog for broader starting pouch item support later
* If enabled, Hylian Shield is always placed in the top-most pouch slot
* Bottles fill the remaining pouch slots
* If there are more starting pouch items than available pouch slots, items are still placed in the pouch inventory and will become immediately available when a pouch upgrade is found (this could be changed to have extra items get sent to the item check but that's a little more work and I want to get feedback first :p)

Tests needed:
* Verify that starting with a full wallet works for all combinations (I'm pretty sure I did this but extra verification would be great)
* Verify you get the correct amount of rupees when obtaining a wallet upgrade
* Verify starting with a shield/bottles works as expected
* Verify I didn't break existing starting health or interface choice stuff
